core/src/test/scala/unit/kafka/server/CreateTopicsRequestTest.scala

```diff
@@ -0,0 +1,212 @@
+      else
+        details.numPartitions
+
+      val replication = if (!details.replicasAssignments.isEmpty)
+        details.replicasAssignments.asScala.head._2.size
+      else
+        details.replicationFactor
+
+      assertNotNull("The topic should be created", metadataForTopic)
+      assertEquals("The topic should have the correct number of partitions", partitions, metadataForTopic.partitionMetadata.size)
+      assertEquals("The topic should have the correct replication factor", replication, metadataForTopic.partitionMetadata.asScala.head.replicas.size)
+    }
+  }
+
+  @Test
+  def testErrorCreateTopicsRequests() {
+    val timeout = 10000
+    val existingTopic = "existing-topic"
+    TestUtils.createTopic(zkUtils, existingTopic, 1, 1, servers)
+
+    // Basic
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map(existingTopic -> new CreateTopicsRequest.TopicDetails(1, 1.toShort)).asJava, timeout),
+      Map(existingTopic -> Errors.TOPIC_ALREADY_EXISTS))
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map("error-partitions" -> new CreateTopicsRequest.TopicDetails(-1, 1.toShort)).asJava, timeout),
+      Map("error-partitions" -> Errors.INVALID_PARTITIONS))
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map("error-replication" -> new CreateTopicsRequest.TopicDetails(1, (numBrokers + 1).toShort)).asJava, timeout),
+      Map("error-replication" -> Errors.INVALID_REPLICATION_FACTOR))
+    val invalidConfig = Map("not.a.property" -> "error").asJava
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map("error-config" -> new CreateTopicsRequest.TopicDetails(1, 1.toShort, invalidConfig)).asJava, timeout),
+      Map("error-config" -> Errors.INVALID_CONFIG))
+    val invalidAssignments = replicaAssignmentToJava(Map(0 -> List(0, 1), 1 -> List(0)))
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map("error-assignment" -> new CreateTopicsRequest.TopicDetails(invalidAssignments)).asJava, timeout),
+      Map("error-assignment" -> Errors.INVALID_REPLICA_ASSIGNMENT))
+
+    // Partial
+    validateErrorCreateTopicsRequests(
+      new CreateTopicsRequest(Map(
+        existingTopic -> new CreateTopicsRequest.TopicDetails(1, 1.toShort),
+        "partial-partitions" -> new CreateTopicsRequest.TopicDetails(-1, 1.toShort),
+        "partial-replication" -> new CreateTopicsRequest.TopicDetails(1, (numBrokers + 1).toShort),
+        "partial-none" -> new CreateTopicsRequest.TopicDetails(1, 1.toShort)).asJava, timeout),
+      Map(
+        existingTopic -> Errors.TOPIC_ALREADY_EXISTS,
+        "partial-partitions" -> Errors.INVALID_PARTITIONS,
+        "partial-replication" -> Errors.INVALID_REPLICATION_FACTOR,
+        "partial-none" -> Errors.NONE
+      )
+    )
+    validateTopicExists("partial-none")
+
+    // Timeout
+    validateErrorCreateTopicsRequests(new CreateTopicsRequest(Map("error-timeout" -> new CreateTopicsRequest.TopicDetails(10, 3.toShort)).asJava, 1),
```

Comment:
I think given all that needs to occur its basically impossible for this to complete within 1ms. I would suspect a bug in the code if it did. 

I cannot have the timeout be zero because that is a separate branch of code where we return immediately and don't even try to wait. (I have a different test for the 0 scenario)